src/main/java/org/mockito/internal/creation/instance/ConstructorInstantiator.java

```diff
@@ -5,26 +5,30 @@
 package org.mockito.internal.creation.instance;
 
 import java.lang.reflect.Constructor;
+import java.util.Arrays;
+
 import org.mockito.internal.util.reflection.AccessibilityChanger;
 
 import static org.mockito.internal.util.StringUtil.join;
 
 public class ConstructorInstantiator implements Instantiator {
 
-    private final Object outerClassInstance;
+    private final Object[] constructorArgs;
+
+    public ConstructorInstantiator(Object... constructorArgs) {
+        if (constructorArgs != null) {
+            this.constructorArgs = constructorArgs;
+        } else {
+            this.constructorArgs = new Object[0];
```

Comment:
Agreed. `ConstructorInstantiatorTest` passes `null` to it in a couple of cases, but you're right - the test should evolve together with the code, especially for internal classes. I'll change the tests to use empty arrays.